motionplan/motionPlanner.go

```diff
@@ -54,10 +54,57 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
+	if req.Logger == nil {
+		req.Logger = golog.NewLogger("plan-request-logger")
+	}
+
+	if req.Frame == nil {
+		return errors.New("PlanRequest cannot have nil frame")
+	}
+
+	if req.FrameSystem == nil {
+		req.Logger.Debug("no FrameSystem was given, so we ephemerally create a frame system containing just the frame for the solve")
+		fs := frame.NewEmptyFrameSystem("")
+		if err := fs.AddFrame(req.Frame, fs.World()); err != nil {
+			return err
+		}
+		req.FrameSystem = fs
+	} else if req.FrameSystem.Frame(req.Frame.Name()) == nil {
+		return frame.NewFrameMissingError(req.Frame.Name())
+	}
+
+	if req.Goal == nil {
+		return errors.New("PlanRequest cannot have nil goal")
+	} else if req.Goal != nil {
+		goalParentFrame := req.Goal.Parent()
+		if req.FrameSystem.Frame(goalParentFrame) == nil {
+			return frame.NewParentFrameMissingError(req.Goal.Name(), goalParentFrame)
+		}
+	}
+
+	if len(req.Frame.DoF()) != len(req.StartConfiguration) {
+		return frame.NewIncorrectInputLengthError(len(req.StartConfiguration), len(req.Frame.DoF()))
+	}
+
+	if req.WorldState == nil {
+		req.Logger.Debug("no WorldState was given, so we created one")
+		wordstate, err := frame.NewWorldState(nil, nil)
+		if err != nil {
+			return err
+		}
+		req.WorldState = wordstate
+	}
+
+	return nil
+}
+
 // PlanMotion plans a motion from a provided plan request.
 func PlanMotion(ctx context.Context, request *PlanRequest) (Plan, error) {
-	if request.Goal == nil {
-		return nil, errors.New("no destination passed to Motion")
+	// make sure request is well formed and not missing vital information
+	if err := request.validatePlanRequest(); err != nil {
```

Comment:
Why are we calling `validatePlanRequest` both in the `PlanMotion` function & in `PlanFrameMotion`? 

Shouldn't this be the responsibility of `PlanMotion` alone?